include/CL/sycl/vec.hpp

```diff
@@ -57,8 +57,13 @@ struct elem {
  */
 template<typename, int>
 class vec;
+
+template <typename DataType, int numElements>
+using __swizzled_vec__ = vec<DataType, numElements>;
```

Comment:
Why defining this if it is only a `vec` ?

Reply:
But it is not clear to me how it can works if it is not a reference or a proxy object implementing `operator=`.

Why is
```
	v.x() = 4;
	v.y() = 5;
	v.z() = 6;
	BOOST_CHECK(v.x() == 4);
	BOOST_CHECK(v.y() == 5);
	BOOST_CHECK(v.z() == 6);
```
working then ? I am puzzled...

Reply:
The whole current __swizzled_vec__ implementation is just a hack, we need to rewrite it to follow the rules in the spec. It doesn't appear at the moment that CTS actually tests it usefully either.

So at the moment consider it a placeholder, I'm defining it just so things build and the API looks sane, but it does need to be a different object, and I'm not sure at what level it needs to be derived from vec or just contain a vec or what.

Reply:
After looking at the latest discussions from Khronos, I have the feeling that implementing `__swizzled_vec__` as a `vec` of element references would just work, as a first step.
Of course compared to a real proxy object we would not bark about things like  `v.xx() = { 2,3 }` but it is not required...